src/Symfony/Component/Console/Helper/ProgressBar.php

```diff
@@ -50,7 +50,8 @@ final class ProgressBar
     private OutputInterface $output;
     private int $step = 0;
     private int $startingStep = 0;
-    private ?int $max = null;
+    private int $max = 0;
+    private bool $isMaxKnown = false;
```

Comment:
Do we need this new variable? I suppose `0` is not a valid max, so having max at 0 probably means that it is not known?

Reply:
the whole point of this PR and the issue I found with Progessbars is that max = 0 IS a valid max value but it is treated like it is not.

we stumbled over that behavior because we need to do a lot of import/sync operations, nasty but necessary. Once in a while, one of our imports/sync iterables is empty and thus max value = 0. `foreach([] as ...)` does not throw errors in vanilla PHP. Using the progressbar with `foreach($bar->iterate([]) as ...)` does.

Reply:
the whole point of this PR and the issue I found with Progessbars is that max = 0 IS a valid max value but it is treated like it is not.

we stumbled over that behavior because we need to do a lot of import/sync operations, nasty but necessary. Once in a while, one of our imports/sync iterables is empty and thus max value = 0. foreach([] as ...) does not throw errors in vanilla PHP. Using the progressbar with foreach($bar->iterate([]) as ...) does.